### PR TITLE
Add cos

### PR DESCRIPTION
**Pull request by jat-canonical**
This feature adds the integration to cos-lite bundle. The integration is handled through a proxy subordinate charm `grafana-agent`. This proxy charm implements `cos-agent` interface and also provides a single relation to handle integrations to `prometheus`, `loki` and `grafana`. The `grafana-agent` charm consumes the cross-model relations from the cos model. The proxy charm relates to the workload app charm as a subordinate to scrape the required information and  to write the metrics, logs and dashboards to the cos charms running in the cos model.

### CODE REVIEW

File: src/ams.py [hunk 1]
```diff
@@ -87,9 +87,25 @@ class PrometheusConfig:
...
 
+    @property
+    def scrape_jobs(self) -> List[Dict]:
+        """Generate scrape jobs for prometheus."""
+        job = {
+            "job_name": "metrics",
+            "metrics_path": self.metrics_path,
+            "static_configs": [{"targets": [f"{self.target_ip}:{self.target_port}"]}],
+        }
+        if self.basic_auth_password and self.basic_auth_password:
```
Comment (adglkh) [inline src/ams.py]: if self.basic_auth_password and self.basic_auth_password:
->
 if self.basic_auth_username and self.basic_auth_password:

File: src/ams.py [hunk 2]
```diff
@@ -87,9 +87,25 @@ class PrometheusConfig:
...
+            auth = {"username": self.basic_auth_username, "password": self.basic_auth_password}
+            job.update(basic_auth=auth)
+        if self.tls_key_path and self.tls_cert_path:
```
Comment (adglkh) [inline src/ams.py]: Check the existence of the key pair path instead?

File: src/ams.py [hunk 3]
```diff
@@ -230,6 +247,7 @@ def _get_config(self) -> dict:
 
     def _set_config_item(self, name, value):
         subprocess.run(["/snap/bin/amc", "config", "set", name, value], check=True)
+        logger.info("Set ams configuration item: %s=%s", name, value)
```
Comment (adglkh) [inline src/ams.py]: This may leak the credential info in the logging system.

**Review by adglkh — CHANGES_REQUESTED** [review]

File: src/ams.py (hunk 3 above)
Comment (jat-canonical) [reply src/ams.py]: yeah, good spot. I wanted it to be debug level anyway but should I just get rid of it or make an exception for a few config items `image.auth` etc.?

**Review by ajanon — APPROVED** [review]
LGTM

File: src/ams.py (hunk 3 above)
Comment (morphis) [reply src/ams.py]: I would leave this out or make it only print the key

**Review by morphis — CHANGES_REQUESTED** [review]

**Review by morphis — APPROVED** [review]
LGTM

**Review by adglkh — APPROVED** [review]
LGTM

**Pull request merged**